app/controllers/api/subcollections/container_images.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Api
+  module Subcollections
+    module ContainerImages
+      def container_images_scan_resource(provider_name, resource_name = nil, image_id = nil, _payload = nil)
+        raise BadRequestError, "Must specify an id for scanning a #{provider_name} resource" unless resource_name
+        image = ContainerImage.find(image_id)
+        image_name = image.name
+        task = nil
+        begin
```

Comment:
You probably don't need the begin statement, and can just rescue the errors similarly to how `scan_container_image` is structured below, with the action results for both success and failure.

Reply:
if i remove the `begin` block and rescue anything, we wind up catching 404s which get converted to 200s. so i think it's correct to leave the begin block in